Wangscape/Reseedable.cpp

```diff
@@ -0,0 +1,285 @@
+    mg->insert("output", makeReseedable(multiply_p));
+    return makeReseedable(mg);
+}
+
+Reseedable Reseedable::operator*(double scale)
+{
+    std::shared_ptr<noise::module::ScaleBias> scale_bias_p = std::make_shared<noise::module::ScaleBias>();
+    scale_bias_p->SetSourceModule(0, *module);
+    scale_bias_p->SetScale(scale);
+
+    std::shared_ptr<ModuleGroup> mg = std::make_shared<ModuleGroup>();
+    mg->insert("source", *this);
+    mg->insert("output", makeReseedable(scale_bias_p));
+    return makeReseedable(mg);
+}
+
+Reseedable Reseedable::pow(Reseedable & exponent)
+{
+    std::shared_ptr<noise::module::Power> power_p = std::make_shared<noise::module::Power>();
+    power_p->SetSourceModule(0, *module);
+    power_p->SetSourceModule(1, *exponent.module);
+
+    std::shared_ptr<ModuleGroup> mg = std::make_shared<ModuleGroup>();
+    mg->insert("base", *this);
+    mg->insert("exponent", exponent);
+    mg->insert("output", makeReseedable(power_p));
+    return makeReseedable(mg);
+}
+
+Reseedable Reseedable::pow(double exponent)
+{
+    std::shared_ptr<Power> exponent_p = std::make_shared<Power>();
+    exponent_p->SetSourceModule(0, *module);
+    exponent_p->SetExponent(exponent);
+
+    std::shared_ptr<ModuleGroup> mg = std::make_shared<ModuleGroup>();
+    mg->insert("source", *this);
+    mg->insert("output", makeReseedable(exponent_p));
+    return makeReseedable(mg);
+}
+
+Reseedable Reseedable::blend(Reseedable& source_a, Reseedable& source_b)
+{
+    std::shared_ptr<noise::module::Blend> blend_p = std::make_shared<noise::module::Blend>();
+    blend_p->SetSourceModule(0, *source_a.module);
+    blend_p->SetSourceModule(1, *source_b.module);
+    blend_p->SetControlModule(*module);
+
+    std::shared_ptr<ModuleGroup> mg = std::make_shared<ModuleGroup>();
+    mg->insert("control", *this);
+    mg->insert("source0", source_a);
+    mg->insert("source1", source_b);
+    mg->insert("output", makeReseedable(blend_p));
+    return makeReseedable(mg);
+}
+
+Reseedable Reseedable::select(Reseedable& source_a, Reseedable& source_b)
+{
+    std::shared_ptr<noise::module::Select> select_p = std::make_shared<noise::module::Select>();
+    select_p->SetSourceModule(0, *source_a.module);
+    select_p->SetSourceModule(1, *source_b.module);
+    select_p->SetControlModule(*module);
+
+    std::shared_ptr<ModuleGroup> mg = std::make_shared<ModuleGroup>();
+    mg->insert("control", *this);
+    mg->insert("source0", source_a);
+    mg->insert("source1", source_b);
+    mg->insert("output", makeReseedable(select_p));
+    return makeReseedable(mg);
+}
+
+Reseedable Reseedable::rotatePoint(double x_angle, double y_angle, double z_angle)
+{
+    std::shared_ptr<noise::module::RotatePoint> rotate_point_p = std::make_shared<noise::module::RotatePoint>();
+    rotate_point_p->SetSourceModule(0, *module);
+    rotate_point_p->SetXAngle(x_angle);
+    rotate_point_p->SetYAngle(y_angle);
+    rotate_point_p->SetZAngle(z_angle);
```

Comment:
`noise::RotatePoint` doesn't support chaining. But using `SetAngles()` would make the function more concise.